src/atoms/UserState.ts

```diff
@@ -1,18 +1,38 @@
 import {atom, selector} from 'recoil';
 import {IUserResource} from 'src/resources/UserResource';
 import {IOrganization} from 'src/resources/OrganisationResource';
+import {getUser} from 'src/resources/UserResource';
 
-// TODO: Error handling
-export const UserState = atom<IUserResource | undefined>({
+// Request ID is used to refresh userState via the API.
+// Updating UserRequestId get's the latest contents of
+// the user state.
+export const UserRequestId = atom({
+  key: 'userRequestId',
+  default: 0,
+});
+
+export const CurrentUsernameState = atom({
+  key: 'currentUsernameState',
+  default: '',
+});
+
+export const UserState = selector<IUserResource | undefined>({
   key: 'userState',
-  default: undefined,
+  get: async ({get}) => {
+    get(UserRequestId); // Add dep on UserRequestId to allow refreshes
+    return await getUser();
+  },
 });
 
 export const UserOrgs = selector<IOrganization[] | undefined>({
   key: 'userOrgs',
   get: async ({get}) => {
-    const user = get(UserState);
-    return user ? user.organizations : undefined;
+    const user: IUserResource = get(UserState);
+    // Check should never fail, if request was unsuccessful it throws
```

Comment:
I feel like when I see this sort of comment, it also means if the error condition does happen it'll be harder to troubleshoot. Do you mind adding something to catch and log something to console at the minimum in the super unlikely event it ever catches?

Reply:
True, added an error log.